modules/lvgl/input/lvgl_encoder_input.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2023 Fabian Blatz <fabianblatz@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT zephyr_lvgl_encoder_input
+
+#include "lvgl_common_input.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(lvgl);
+
+struct lvgl_encoder_input_config {
+	struct lvgl_common_input_config common_config; /* Needs to be first member */
+	const uint16_t *rotation_input_codes;
+	uint8_t rotation_num_codes;
+	const uint16_t *button_input_codes;
+	uint8_t button_num_codes;
+};
+
+static void lvgl_encoder_process_event(const struct device *dev, struct input_event *evt)
+{
+	struct lvgl_common_input_data *data = dev->data;
+	const struct lvgl_encoder_input_config *cfg = dev->config;
+	bool event_processed = false;
+	uint8_t i;
```

Comment:
well the API only support one axis and one button right? I'd say change the properties from `array` to `int` and then drop the num_codes and these cycles, make things a bit simpler. Guess you could make the button optional, default it to -1 and use that to see if it's present, keep rotation mandatory (no point having the node without it)

Reply:
Yeah my thought was that maybe you can map multiple physical qdecs to one lvgl encoder indev, but I guess this would produce odd behaviour in practice.